examples/ex40.cpp

```diff
@@ -0,0 +1,350 @@
+//                                MFEM Example 40
+//
+// Compile with: make ex40
+//
+// Sample runs: ex40 -step 10 -gr 2.0
+//              ex40 -step 10 -gr 2.0 -o 3 -r 1
+//              ex40 -step 10 -gr 2.0 -r 4 -m ../data/l-shape.mesh
+//
+// Description: This example code demonstrates how to use MFEM to solve the
+//              eikonal equation,
+//
+//                      |∇u| = 1 in Ω,  u = g on ∂Ω.
+//
+//              This example constructs a fast converging sequence,
+//
+//                      uₖ → u  as k → \infty,
```

Comment:
Can this method work with `u=g` at a point or a curve inside the domain?

Reply:
The method is not designed for implicitly defined surfaces, but I also have an idea for a way to do that. @alizma and I could work on it over the summer